sources/api/datastore/src/filesystem.rs

```diff
@@ -489,6 +498,95 @@ impl DataStore for FilesystemDataStore {
         let pending = Committed::Pending {
             tx: transaction.into(),
         };
+
+        // We will first commit metadata to correctly check that if a setting exist and
+        // its strength is strong, we will not change it to weak.
+        // Get metadata from pending transaction
+        let transaction_metadata =
+            self.get_metadata_prefix("settings.", &pending, &None as &Option<&str>)?;
+
+        trace!(
+            "commit_transaction: transaction_metadata: {:?}",
+            transaction_metadata
+        );
+
+        for (key, value) in transaction_metadata {
+            for (metadata_key, metadata_value) in value {
+                // For now we are only processing the strength metadata from pending
+                // transaction to live
+                if metadata_key.name() != "strength" {
+                    continue;
+                }
+
+                // strength in pending transaction
+                let pending_strength: String =
+                    deserialize_scalar::<_, ScalarError>(&metadata_value.clone())
+                        .with_context(|_| error::DeSerializeSnafu {})?;
+
+                // Get the setting strength in live
+                // get_metadata function returns Ok(None) in case strength does not exist
+                // We will consider this case as strength equals strong.
+                let committed_strength =
+                    match self.get_metadata(&metadata_key, &key, &Committed::Live) {
+                        Ok(Some(v)) => v,
+                        Ok(None) => "strong".to_string(),
+                        Err(_) => continue,
+                    };
```

Comment:
We need to bubble up errors here rather than dropping them. This incidentally makes it a bit easier to handle the "missing" case.


```suggestion
                let committed_strength = match self.get_metadata(&metadata_key, &key, &Committed::Live)?.unwrap_or_else(|| "strong".to_string());
```

This would also be a good time to deserialize the value into the `enum` that you've defined in `models`, e.g. like:

```suggestion
                let committed_strength: Strength = match self.get_metadata(&metadata_key, &key, &Committed::Live)?
                    .map(Strength::parse)
                    .transpose()?
                    .unwrap_or_default();
```

Reply:
I am unsure if we need to error out in between, because suppose we are running this for multiple setting metadata, we will commit the strength for some of them and leave some from the transaction. We will also leave the settings un-committed in that case. This can leave the system in weird state.

Reply:
We can resolve this by splitting the transaction into two phases -- an information gathering/decision phase, and a write phase. In the first phase we merely process the incoming write and make a decision on whether or not we can proceed.

Afterwards, we proceed to write the transaction.